qiskit/algorithms/eigensolvers/ssvqe.py

```diff
@@ -0,0 +1,575 @@
+    if the ansatz returns ``None`` as the upper bound, the default value will be :math:`2\pi`.
+
+    An optional list of initial states, via the *initial_states*, may also be provided.
+    Choosing these states appropriately is a critical part of the algorithm. They must
+    be mutually orthogonal because this is how the algorithm enforces the mutual
+    orthogonality of the solution states. If the *initial_states* is left as ``None``,
+    then SSVQE will automatically generate a list of computational basis states and use
+    these as the initial states. For many physically-motivated problems, it is advised
+    to not rely on these default values as doing so can easily result in an unphysical
+    solution being returned. For example, if one wishes to find the low-lying excited
+    states of a molecular Hamiltonian, then we expect the output states to belong to a
+    particular particle-number subspace. If an ansatz that preserves particle number
+    such as :class:`UCCSD` is used, then states belonging to the incorrect particle
+    number subspace will be returned if the *initial_states* are not in the correct
+    particle number subspace. A similar statement can often be made for the
+    spin-magnetization quantum number.
+
+    The following attributes can be set via the initializer but can also be read and
+    updated once the SSVQE object has been constructed.
+
+    Attributes:
+            estimator (BaseEstimator): The primitive instance used to perform the expectation
+                estimation of observables.
+            k (int): The number of eigenstates that SSVQE will attempt to find.
+            ansatz (QuantumCircuit): A parameterized circuit used as an ansatz for the
+                wave function.
+            optimizer (Optimizer): A classical optimizer, which can be either a Qiskit optimizer
+                or a callable that takes an array as input and returns a Qiskit or SciPy optimization
+                result.
+            gradient (BaseEstimatorGradient | None): An optional estimator gradient to be used with the
+                optimizer.
+            callback (Callable[[int, np.ndarray, Sequence[float], dict[str, Any]], None] | None): A
+                function that can access the intermediate data at each optimization step. These data are
+                the evaluation count, the optimizer parameters for the ansatz, the evaluated mean
+                energies, and the metadata dictionary.
+            weight_vector (Sequence[float]): A 1D array of real positive-valued numbers to assign
+                as weights to each of the expectation values. If ``None``, then SSVQE will default
+                to [k, k-1, ..., 1].
+            initial_states (Sequence[QuantumCircuit]): An optional list of mutually orthogonal
+                initial states. If ``None``, then SSVQE will set these to be a list of mutually
+                orthogonal computational basis states.
```

Comment:
Perhaps it might be nicer to maintain the order of these arguments as in the init.

Reply:
I agree. I have now changed this.